Assets/scripts/Enemy/View/ViewEnemy.cs

```diff
@@ -30,7 +30,7 @@ public Vector3 GetPosition()
     }
     public void TakeDamage(int damage,IBasePlayerController player)
     {
-        controller.BulletHit(damage,player);
+        controller.BulletHit(damage,(ControllerPlayer)player);
```

Comment:
point of interfaces is that you don't need to know about controllerplayer implementation in view enemy but now you have this hard coupled. work with interfaces